services/src/integ-test/java/uk/gov/moj/sdt/services/messaging/IndividualRequestMdbIntTest.java

```diff
@@ -30,76 +30,96 @@
 import org.springframework.test.context.ActiveProfiles;
 import org.springframework.test.context.jdbc.Sql;
-import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.context.jdbc.Sql;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.springframework.transaction.annotation.Transactional;
+import uk.gov.moj.sdt.services.api.ITargetApplicationSubmissionService;
 import uk.gov.moj.sdt.services.config.ServicesTestConfig;
-import uk.gov.moj.sdt.services.messaging.api.IMessageWriter;
-import uk.gov.moj.sdt.services.messaging.api.ISdtMessage;
 import uk.gov.moj.sdt.test.utils.AbstractIntegrationTest;
 import uk.gov.moj.sdt.test.utils.TestConfig;
 
-import java.io.IOException;
+import javax.jms.JMSException;
+import javax.jms.ObjectMessage;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
 
 /**
  * IntegrationTest class for testing the MessageReader implementation.
  *
  * @author Manoj Kulkarni
  */
 @ActiveProfiles("integ")
-@RunWith(SpringRunner.class)
-@SpringBootTest(classes = {TestConfig.class, ServicesTestConfig.class })
+@ExtendWith(SpringExtension.class)
+@SpringBootTest(classes = {TestConfig.class, ServicesTestConfig.class})
 @Sql(scripts = {"classpath:uk/gov/moj/sdt/services/sql/RefData.sql", "classpath:uk/gov/moj/sdt/services/sql/IndividualRequestMdbIntTest.sql"})
+@Transactional
 public class IndividualRequestMdbIntTest extends AbstractIntegrationTest {
+
+    @Autowired
+    @Qualifier("messageListenerContainer")
+    private DefaultMessageListenerContainer mcolsContainer;
+
+    @Autowired
+    @Qualifier("messageListenerContainerMCol")
+    private DefaultMessageListenerContainer mcolContainer;
+
+    @MockBean
+    private ITargetApplicationSubmissionService targetApplicationSubmissionService;
+
+    private IndividualRequestMdb individualRequestMdb;
+
     /**
      * Setup the test.
      */
-    @Before
+    @BeforeEach
     public void setUp() {
-        // Write a Message to the MDB
-        final ISdtMessage sdtMessage = new SdtMessage();
-        sdtMessage.setSdtRequestReference("SDT_REQ_TEST_1");
-        sdtMessage.setMessageSentTimestamp(System.currentTimeMillis());
-        sdtMessage.setEnqueueLoggingId(1);
-        final IMessageWriter messageWriter = (IMessageWriter) this.applicationContext.getBean("MessageWriter");
-        messageWriter.queueMessage(sdtMessage, "MCOLS", false);
+        individualRequestMdb = new IndividualRequestMdb(targetApplicationSubmissionService);
     }
 
     /**
      * This method tests the read message.
      *
      * @throws InterruptedException if the thread call is interrupted
-     * @throws IOException          if there is any problem when reading the file
+     * @throws JMSException         if there is any problem when reading the file
```

Comment:
Is the description correct?  Not sure if a JMSException would occur when reading a file.

Reply:
done